File: backend/db/models/step.js
"use strict";
const { Model } = require("sequelize");
module.exports = (sequelize, DataTypes) => {
  class Step extends Model {
    /**
     * Helper method for defining associations.
     * This method is not a part of Sequelize lifecycle.
     * The `models/index` file will call this method automatically.
     */
    static associate(models) {
      Step.belongsTo(models.Project, {
        as: "steps",
        foreignKey: "projectId",
      });
    }
  }
  Step.init(
    {
      title: DataTypes.STRING(50),
      description: DataTypes.TEXT,
      media: DataTypes.ARRAY(DataTypes.STRING),
      projectId: {
        type: DataTypes.INTEGER,
        references: { model: "Projects" },
      },
    },
    {
      sequelize,
      modelName: "Step",
    }
  );
  return Step;
};

File: backend/app/models/step.py
# pylint: disable=too-few-public-methods
from .db import db


class Step(db.Model):
    __tablename__ = 'steps'
    
    id = db.Column(db.Integer, primary_key=True)
    title = db.Column(db.String(20))
    description = db.Column(db.Text)
    media = db.Column(db.ARRAY(db.String))
    project_id = db.Column(db.Integer, db.ForeignKey('projects.id'), nullable=False)
    created_at = db.Column(db.DateTime, default=db.func.now())
    updated_at = db.Column(db.DateTime, default=db.func.now(), onupdate=db.func.now())

    def to_dict(self):
        return {
            "id": self.id,
            "title": self.title,
            "description": self.description,
            "media": self.media,
        }
